src/test/codegen/naked-functions.rs

```diff
@@ -9,6 +9,7 @@
 #[naked]
 pub fn naked_empty() {
     // CHECK-NEXT: {{.+}}:
+    // CHECK-NEXT: %0 = alloca {}, align 1
```

Comment:
I feel like this is probably invalid – for `#[naked]` functions to be well-formed, they usually should only contain an optional inline assembly call and ret. `alloca` here will likely make any use of `#[naked]` plain unsound.

 If `#[naked]` wasn't as unstable as they are, this would be blocking, and IMHO it needing change now warrants at least a folllow-up issue.

Reply:
FWIW I'm pretty sure we skip codegenning ZST assignments/allocas/etc in the llvm backend which is why with `_0 = const ()` in MIR it never generated any LLVM. Now that there are no assignments to the `_0`, its probably taking a different path in the backend and ends up generating the alloca anyway.

If I had to guess this local is failing to be marked as a SSA value?

Reply:
So this pr should be blocked until the codegen is fixed to not generate ZST allocas for _0 if it is not assigned in MIR?

I haven't looked at codegen before, so any pointers are appreciated.

Reply:
Yeah, that looks like a use of an uninitialized ZST local marking it as non-SSA (implicit use in return terminator?). That part would be in `non_ssa_locals`.

Reply:
The latest commit tries to eliminate the alloca